pkg/driverbuilder/builder/builders.go

```diff
@@ -106,6 +113,132 @@ func Script(b Builder, c Config, kr kernelrelease.KernelRelease) (string, error)
 	return buf.String(), nil
 }
 
+type GCCVersionRequestor interface {
+	GCCVersion(kr kernelrelease.KernelRelease) float64
+}
+
+type Image struct {
+	GCCVersion []float64
+}
+
+var images = map[string]Image{
+	"buster": {
+		GCCVersion: []float64{4.8, 5, 6, 8},
+	},
+	"bullseye": {
+		GCCVersion: []float64{9, 10},
+	},
+	"bookworm": {
+		GCCVersion: []float64{11, 12},
+	},
+}
+
+func defaultGCC(kr kernelrelease.KernelRelease) float64 {
+	switch kr.Version {
+	case 5:
+		if kr.PatchLevel >= 18 {
+			return 12
+		}
+		return 11
+	case 4:
+		return 8
+	case 3:
+		return 5
+	case 2:
+		return 4.8
+	default:
+		return 12
+	}
+}
+
+func (b *Build) SetGCCVersion(builder Builder, kr kernelrelease.KernelRelease) {
+	if b.GCCVersion != 0 {
+		// If set from user, go on
+		return
+	}
+
+	b.GCCVersion = 8 // default value
+
+	distance := math.MaxFloat64
+
+	// if builder implements "GCCVersionRequestor" interface -> use it
+	// Else, fetch the best builder available from the kernelrelease version
+	// using the deadly simple defaultGCC() algorithm
+	// Always returns the nearest one
+	var targetGCC float64
+	if bb, ok := builder.(GCCVersionRequestor); ok {
+		targetGCC = bb.GCCVersion(kr)
+	} else {
+		targetGCC = defaultGCC(kr)
+	}
+
+	for name, img := range images {
+		var foundGCC float64
+		d := math.MaxFloat64
+		for _, gcc := range img.GCCVersion {
+			if math.Abs(gcc-targetGCC) < d {
+				// Find the nearest to targetGCC
```

Comment:
Hmm, shouldn't this be the next available `gcc` version, rather than the nearest? For instance, if we have a `targetGCC` of 5.1, I'd think that 6 would be the version that should get chosen rather than 5?

Reply:
I second this question

Reply:
Yeah, i am not sure if it makes any difference: i think that a target gcc of let's say 5.4 will always be buildable from a gcc 5.0 let's say.
But we can easily make the change here if you wish :)

Reply:
I wanted to avoid cases where a GCC X is too new to build a GCC Y target, ie:
* target is 6.5
* we have 6.3 or 9
* we chose 9
* 9 does not build because it is too new (glibc mismatch)

Reply:
That makes sense to me, thank you for explaining!